status/reporter_github.go

```diff
@@ -238,19 +238,32 @@ func (cru *CheckRunStatusUpdater) UpdateStatus(ctx context.Context, report TestR
 		if err != nil {
 			cru.logger.Error(err, "failed to create checkrun",
 				"checkRunAdapter", checkRunAdapter)
-			return err
 		}
-	} else {
-		cru.logger.Info("found existing checkrun", "existingCheckRun", existingCheckrun)
-		err = cru.ghClient.UpdateCheckRun(ctx, *existingCheckrun.ID, checkRunAdapter)
+		return err
+	}
+
+	cru.logger.Info("found existing checkrun", "existingCheckRun", existingCheckrun)
+
+	// If pre-existing checkrun is already completed, then create a
+	// new checkrun with same external ID, rather than updating it
+	if existingCheckrun.GetStatus() == "completed" {
+		cru.logger.Info("The existing checkrun is already in completed state, re-creating a new checkrun for scenario test status of snapshot",
+			"snapshot.NameSpace", cru.snapshot.Namespace, "snapshot.Name", cru.snapshot.Name, "scenarioName", report.ScenarioName)
+		_, err = cru.ghClient.CreateCheckRun(ctx, checkRunAdapter)
 		if err != nil {
-			cru.logger.Error(err, "failed to update checkrun",
+			cru.logger.Error(err, "failed to create checkrun",
 				"checkRunAdapter", checkRunAdapter)
-			return err
 		}
+		return err
+	}
 
+	err = cru.ghClient.UpdateCheckRun(ctx, *existingCheckrun.ID, checkRunAdapter)
+	if err != nil {
+		cru.logger.Error(err, "failed to update checkrun",
```

Comment:
Would it be more convenient to add existingCheckrun.ID also into the error message to better traceability? Feel free to prove me wrong, I don't know what checkrunAdapter information consist of.

Reply:
Hmm good point, I think that since the checkrun's ID field is just a bunch of numbers (example: `ID:26087296145`) it's not very helpful while debugging. We can instead use ExternalID field as that is a combination of scenario-name + component-name and it is unique for each checkrun. It's already a part of `checkrunAdapter` ([here](https://github.com/konflux-ci/integration-service/blob/be17a66db2d4ac7fc7ed676b35158a8bd8ccc4e8/git/github/github.go#L38)), that gets printed within the above error message.

Also, we're printing the ExternalID again [here](https://github.com/konflux-ci/integration-service/blob/be17a66db2d4ac7fc7ed676b35158a8bd8ccc4e8/git/github/github.go#L381), and then the entire existing checkrun object [here](https://github.com/konflux-ci/integration-service/pull/770/files#diff-83bd1d9146518ccfb3036c984464501bca2c88acc23ffcd801c9057e7c0df48eR245). So, we'd be able to compare both existing checkrun and the checkrun we're trying to create/update, in any case.

Reply:
I think that the `externalID` is the main info that we use for debugging so I think that's enough for our needs.